modules/standard/Path.chpl

```diff
@@ -457,4 +458,182 @@ proc file.realPath(): string throws {
      return ("", name);
    }
  }
+
+ /*
+
+ Returns the parent directory of the :type:`~IO.file` record.  For instance,
+ a file with path `/foo/bar/baz` would return `/foo/bar`
+
+  :return: The parent directory of the file
+  :rtype: `string`
+
+ */
+ pragma "no doc"
+ proc file.getParentName(out error:syserr): string {
+   check(error);
+
+   var ret: string = "unknown";
+   if !error {
+     var tmp: string;
+     tmp = this.realPath(error);
+     if !error then
+       ret = dirname(new string(tmp));
+   }
+   return ret;
+ }
+
+ /*
+ Returns the parent directory of the :type:`~IO.file` record.  For instance,
+ a file with path `/foo/bar/baz` would return `/foo/bar`
+
+ Will halt with an error message if one is detected.
+
+  :return: The parent directory of the file
+  :rtype: `string`
+ */
+ proc file.getParentName(): string throws {
```

Comment:
Something's gone awry with your merge.  There are now two of each of the ``proc file.getParentName()`` definitions, and two ``joinPath()``s and two ``isAbsPath()``s.  :(

Reply:
Thanks for pointing that out. I resolved it!